x-pack/plugin/mapper-aggregate-metric/src/main/java/org/elasticsearch/xpack/aggregatemetric/mapper/AggregateDoubleMetricFieldMapper.java

```diff
@@ -605,24 +606,21 @@ protected void parseCreateField(DocumentParserContext context) throws IOExceptio
                 // new aggregate metric types are added (histogram, cardinality etc)
                 ensureExpectedToken(XContentParser.Token.VALUE_NUMBER, token, subParser);
                 NumberFieldMapper delegateFieldMapper = metricFieldMappers.get(metric);
-                // We don't accept arrays of metrics
-                if (context.doc().getField(delegateFieldMapper.fieldType().name()) != null) {
-                    throw new IllegalArgumentException(
-                        "Field ["
-                            + name()
-                            + "] of type ["
-                            + typeName()
-                            + "] does not support indexing multiple values for the same field in the same document"
-                    );
-                }
                 // Delegate parsing the field to a numeric field mapper
-                delegateFieldMapper.parse(context);
-
+                try {
+                    delegateFieldMapper.parse(context);
+                } catch (MapperParsingException e) {
+                    if (e.getCause() instanceof IllegalArgumentException) {
+                        throw (IllegalArgumentException) e.getCause();
+                    } else {
+                        throw e;
+                    }
```

Comment:
I guess that you extract`IllegalArgumentException` from this `MapperParsingException` because you want to bubble up the underlying error instead of throwing a generic message like this `"failed to parse field [field.value_count] of type [integer] in document with id '1'.".

If that's the case, can you please add a comment here to document the thinking behind this?